security.md

```diff
@@ -4,35 +4,39 @@ title: Reporting a Security Vulnerability
+- [Reporting a vulnerability](#reporting-a-vulnerability)
+- [Handling a vulnerability](#handling-a-vulnerability)
 
-This is a private mailing list where only members of the WSO2 internal security team are subscribed to, and is treated as top priority.
+## Reporting a vulnerability
 
-If you wish to send secure messages to [security@ballerina.io](mailto:security@ballerina.io), you may use the following key:
+Ensure you are using the latest Ballerina version before you run an automated security scan or perform a penetration test against them.
 
-security@ballerina.io: 0168 DA26 2989 0DB9 4ACD  8367 E683 061E 2F85 C381 [pgp.mit.edu](https://pgp.surfnet.nl/pks/lookup?op=vindex&fingerprint=on&search=0xE683061E2F85C381)
+If you have any concerns regarding the security aspects of the source code or any other resource in this repo or have uncovered a security vulnerability, we strongly encourage you to report that to our private and highly confidential security mailing list: **[security@ballerina.io](mailto:security@ballerina.io)** first using the below key without disclosing them in any forums, sites, or other groups - public or private. 
 
-## Vulnerability Information
+security@ballerina.io: 0168 DA26 2989 0DB9 4ACD 8367 E683 061E 2F85 C381 [pgp.mit.edu](https://pgp.surfnet.nl/pks/lookup?op=vindex&fingerprint=on&search=0xE683061E2F85C381)
 
-Please use the following template in reporting vulnerabilities:
+We will keep you informed of the progress towards a fix and disclosure of the vulnerability if reported issue is identified as a true positive. To protect the end-user security, these issues could be disclosed in other places only after it’s mitigation actions and disclosure process are completed.
 
-- Vulnerable Ballerina artifacts(s) and version(s)
+**Warning:** Please do not create GitHub issues for security vulnerabilities. Further, kindly refrain from sharing the vulnerability details you come across with other individuals. 
+
+Also, use the following template when reporting vulnerabilities so that it contains all the required information and helps expedite the analysis and mitigation process.
+
+- Vulnerable Ballerina artifact(s) and version(s)
 - Overview: High-level overview of the issue and self-assessed severity
 - Description: Include the steps to reproduce
 - Impact: Self-assessed impact
 - Solution: Any proposed solution
 
-## Vulnerability Handling
+## Handling a vulnerability
+
+The below is an overview of the vulnerability handling process.
 
-Here's an overview of our vulnerability handling process:
+1. The user privately reports the vulnerability to security@ballerina.io. (The initial response time will be less than 24 hours).
+2. The WSO2 security team works privately with the user to fix the vulnerability and QA verifies the solution.
+3. Apply the fix to the master branch and release a new version of the distribution if required.
+4. The reported user is kept updated on the progress of the process. 
```

Comment:
The first two sentences seem to be phrased in a particular manner while the last two sentences have broken away from that pattern. The first two are formed like "A does X". Better to follow the same form for the last two sentences or just not stick to a pattern at all.

Reply:
Changed as follows:

1. The vulnerability will be reported privately to security@ballerina.io. (The initial response time will be less than 24 hours).
2. The reported vulnerability gets fixed and the solution gets verified by the relavant teams at WSO2.
3. The fix gets applied to the master branch and a new version of the distribution gets released if required.
4. The reported user is kept updated on the progress of the process.